src/util/samplebuffer.cpp

```diff
@@ -12,6 +11,17 @@ SampleBuffer::SampleBuffer(SINT size)
 
 SampleBuffer::~SampleBuffer() {
     SampleUtil::free(m_data);
+
+    /* cleanup the memory, just in case of use after free */
+    m_data = nullptr;
+    m_size = 0;
+}
+
+SampleBuffer& SampleBuffer::operator=(SampleBuffer&& that) {
```

Comment:
What is the use case for this? Do we also want a move constructor?

Reply:
A simple use case:

```C++
class MyProcessor
{
   SampleBuffer m_b;
   
   void activate(double samplerate, int blockSize)
   {
      m_b = SampleBuffer(2 * blockSize);
   }
};

Reply:
There is already a move ctor.